Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/GenericConstraintCorrection.h

```diff
@@ -59,9 +59,30 @@ class SOFA_COMPONENT_CONSTRAINT_LAGRANGIAN_CORRECTION_API GenericConstraintCorre
 
     void computeResidual(const core::ExecParams* params, linearalgebra::BaseVector *lambda) override;
 
+    SingleLink<GenericConstraintCorrection, sofa::core::behavior::LinearSolver, BaseLink::FLAG_STOREPATH | BaseLink::FLAG_STRONGLINK> l_linearSolver; ///< Link towards the linear solvers used to compute the compliance matrix, requiring the inverse of the linear system matrices
+    SingleLink<GenericConstraintCorrection, sofa::core::behavior::OdeSolver, BaseLink::FLAG_STOREPATH | BaseLink::FLAG_STRONGLINK> l_ODESolver; ///< Link towards the ODE solver used to recover the integration factors
+    Data< double > d_complianceFactor; ///< Factor applied to the position factor and velocity factor used to calculate compliance matrix.
+
+
+    SOFA_ATTRIBUTE_DEPRECATED("v22.12 (#3152)", "v23.06", "String data of the GenericConstraintCorrection were replaced by explicit links")
     Data< type::vector< std::string > >  d_linearSolversName; ///< name of the constraint solver
+    SOFA_ATTRIBUTE_DEPRECATED("v22.12 (#3152)", "v23.06", "String data of the GenericConstraintCorrection were replaced by explicit links")
     Data< std::string >                    d_ODESolverName; ///< name of the ode solver
-    Data< double > d_complianceFactor; ///< Factor applied to the position factor and velocity factor used to calculate compliance matrix.
+
+    //SOFA_ATTRIBUTE_DEPRECATED("v22.12 (#3152)", "v23.06", "String data of the GenericConstraintCorrection were replaced by explicit links")
+    void parse( sofa::core::objectmodel::BaseObjectDescription* arg ) override
+    {
+        Inherit1::parse(arg);
+        if (arg->getAttribute("solverName"))
+        {
+            msg_warning() << "String data \"solverName\" is now replaced by explicit data link: \"linearSolver\" (PR #3152)";
```

Comment:
Do we want to help the user by automatically setting `l_linearSolver` based on `solverName`?